### PR TITLE
Add capability to run with ilmerge bundling.

### PR DESCRIPTION
**Pull request by stephanemartin**
For my tosca plugin, I need to merge all dependencies to avoid conflict with different version of Odata.  So I've modified your project to look at executed assembly if adapter v3/4 dll was not found.

### CODE REVIEW

File: src/Simple.OData.Client.Core/Adapter/ODataAdapterFactory.cs [hunk 1]
```diff
@@ -127,15 +126,26 @@ private IODataModelAdapter LoadModelAdapter(ITypeCache typeCache, string modelAd
...
+            try
+            {
+                var assemblyNameObj = new AssemblyName(assemblyName);
+                return Assembly.Load(assemblyNameObj).GetType(typeName);
+            }catch(FileNotFoundException fileNotFoundException)
+            {
+                var type = Assembly.GetExecutingAssembly().GetType(typeName);
+                if (type == null)
+                {
+                    throw fileNotFoundException;
```
Comment (watfordgnf) [inline src/Simple.OData.Client.Core/Adapter/ODataAdapterFactory.cs]:
```suggestion
                    throw;
```

File: src/Simple.OData.Client.Core/Adapter/ODataAdapterFactory.cs [hunk 2]
```diff
@@ -127,15 +126,26 @@ private IODataModelAdapter LoadModelAdapter(ITypeCache typeCache, string modelAd
...
-        private Assembly LoadAdapterAssembly(string modelAdapterAssemblyName)
+        private Type LoadType(string assemblyName, string typeName)
         {
-            var assemblyName = new AssemblyName(modelAdapterAssemblyName);
-            return Assembly.Load(assemblyName);
+            try
+            {
+                var assemblyNameObj = new AssemblyName(assemblyName);
+                return Assembly.Load(assemblyNameObj).GetType(typeName);
+            }catch(FileNotFoundException fileNotFoundException)
```
Comment (watfordgnf) [inline src/Simple.OData.Client.Core/Adapter/ODataAdapterFactory.cs]:
```suggestion
            }
            catch(FileNotFoundException fileNotFoundException)
```

File: src/Simple.OData.Client.Core/Adapter/ODataAdapterFactory.cs [hunk 3]
```diff
@@ -127,15 +126,26 @@ private IODataModelAdapter LoadModelAdapter(ITypeCache typeCache, string modelAd
...
+            {
+                var type = Assembly.GetExecutingAssembly().GetType(typeName);
```
Comment (watfordgnf) [inline src/Simple.OData.Client.Core/Adapter/ODataAdapterFactory.cs]: Why not `Type.GetType(typeName)` ?

Comment (stephanemartin) [reply src/Simple.OData.Client.Core/Adapter/ODataAdapterFactory.cs]: Indeed, it's better. I do not know this shortcut thank you.

File: src/Simple.OData.Client.Core/Adapter/ODataAdapterFactory.cs (hunk 2 above)
Comment (stephanemartin) [reply src/Simple.OData.Client.Core/Adapter/ODataAdapterFactory.cs]: We can remove fileNotFoundException now.

File: src/Simple.OData.Client.Core/Adapter/ODataAdapterFactory.cs (hunk 1 above)
Comment (stephanemartin) [reply src/Simple.OData.Client.Core/Adapter/ODataAdapterFactory.cs]: Yes

**Comment by object** [comment]
Thanks for the PR, merged.

**Pull request merged**